cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorUtils.java

```diff
@@ -130,4 +143,15 @@ public static boolean hasUnfixableGoals(GoalViolations goalViolations) {
     List<String> unfixableGoals = goalViolations.violatedGoalsByFixability().get(false);
     return unfixableGoals != null && !unfixableGoals.isEmpty();
   }
+
+  /**
+   * Check whether the given goal violations has unfixable goals.
+   *
+   * @param goalViolations Goal violations to check whether there are unfixable goals.
+   * @return True if the given goal violations contain unfixable goals, false otherwise.
+   */
+  public static boolean hasUnfixableGoals(IntraBrokerGoalViolations goalViolations) {
```

Comment:
Is that better to change the input param to type AbstractGoalViolations? That way this method can help to check both interBrokerGoals and intraBrokerGoals